girder/api/rest.py

```diff
@@ -776,25 +776,46 @@ def removeRoute(self, method, route, handler=None, resource=None):
         :type route: tuple[str]
         :param handler: The method called for the route; this is necessary to
                         remove the documentation.
-        :type handler: function
+        .. deprecated :: 2.3.0
+        :type handler: Function
         :param resource: the name of the resource at the root of this route.
         """
         self._ensureInit()
+
         nLengthRoutes = self._routes[method.lower()][len(route)]
-        for i in range(len(nLengthRoutes)):
-            if nLengthRoutes[i][0] == route:
+        for i, (registeredRoute, registeredHandler) in enumerate(nLengthRoutes):
+            if registeredRoute == route:
+                handler = registeredHandler
                 del nLengthRoutes[i]
                 break
+
         # Remove the api doc
-        if resource is None and hasattr(self, 'resourceName'):
-            resource = self.resourceName
-        elif resource is None:
-            resource = handler.__module__.rsplit('.', 1)[-1]
-        if handler and getattr(handler, 'description', None) is not None:
+        if resource is None:
+            resource = getattr(self, 'resourceName', handler.__module__.rsplit('.', 1)[-1])
+        if getattr(handler, 'description', None) is not None:
             docs.removeRouteDocs(
                 resource=resource, route=route, method=method,
                 info=handler.description.asDict(), handler=handler)
 
+    def getRouteHandler(self, method, route):
+        """
+        Get the handler method for a given method and route.
+
+        :param method: The HTTP method, e.g. 'GET', 'POST', 'PUT'
+        :type method: str
+        :param route: The route, as a list of path params relative to the
+                      resource root, exactly as it was passed to the ``route`` method.
+        :type route: tuple[str]
+        :returns: The handler method for the route.
+        :rtype: Function
+        :raises: `RestException`, when no route can be matched.
+        """
+        for registeredRoute, registeredHandler in self._routes[method.lower()][len(route)]:
+            if registeredRoute == route:
+                return registeredHandler
+        else:
+            raise RestException('Could not find route "%s %s"' % (method.upper(), '/'.join(route)))
```

Comment:
I'm aware of the theoretical taxonomy of Girder's exceptions (though I take issue with the fact that they don't all inherit from a common base class), but as a practical matter, I was just pointing out that it's not followed consistently throughout the codebase.

I'd be happy if we had a family of more specific exceptions, and if existing code was updated to use them properly, but I'm concerned that adding a whole new exception type to serve a single expert-level method which nobody is using yet is a bit outside the scope of this PR.

Perhaps we could change this to throw an `Exception`, then add a more specific type in the future, when we have a chance to consider the taxonomy in a more comprehensive way. Compatibility will be maintained, since everything inherits from `Exception`.